addons/mail/static/src/components/thread_needaction_preview/thread_needaction_preview_tests.js

```diff
@@ -201,6 +201,56 @@ QUnit.test('click on expand from chat window should close the chat window and op
     );
 });
 
+QUnit.test('[technical] opening a non-channel chat window should not call channel_fold', async function (assert) {
+    // channel_fold should not be called when opening non-channels in chat
+    // window, because there is no server sync of fold state for them.
+    assert.expect(3);
+
+    await this.start({
+        hasChatWindow: true,
+        async mockRPC(route, args) {
+            if (route.includes('channel_fold')) {
+                const message = "should not call channel_fold when opening a non-channel chat window";
+                assert.step(message);
+                console.error(message);
+                throw Error(message);
+            }
```

Comment:
just the `throw Error()` looks useful to me

Reply:
Here is not inside the test itself, it's inside the RPC. Errow in RPC is allowed/ignored by qunit.
Only an `assert` can do what we need here. All others are useless.

Reply:
If you don't want the `step` error to propagate to the end of the test, I just need to add an empty `verifyStep` after the step that has to check it.

Reply:
Strange behaviour that such errors are silently ignored by test env, but that's an issue out of scope of this PR.
I'm good to keep this just to help us debug best.